### ISSUE TITLE
{Linked:Bug11228198} [Adaptive Cards > Designer] - Screen Reader is not announcing the descriptive information while navigating to the 'Next or Previous' button.

### ISSUE THREAD

**Issue by vagpt** [microsoft/AdaptiveCards#8762]
### Target Platforms

Other

### SDK Version

1.6

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
URL: https://adaptivecards.io/designer/
OS Version: 23H2 (OS Build 25992.1000)
Browser Version: Version 121.0.2220.3 (Official build) dev (64-bit)
Screen Reader: Narrator

**Pre-Requisite:**
Run the below attached JSON file under 'Copy Card Payload editor.'

**Repro Steps:**
1. Open the above URL on the EDGE Dev browser.
2. Run the Json file.
3. Navigate to the Preview mode and press enter key to activate it.
4. Press tab key and navigate to 'Next and Previous' button and observe the issue.

**Actual Result:**
While navigating to 'Next or Previous' button, screen reader is not announcing descriptive information for the same. It is announcing as 'Next button' or 'Previous' button.
Also, after activating next or previous button screen reader is not conveying the descriptive information about the changes. It is not announcing the information about the changed slide.

**Expected Result:**
Screen Reader should announce the descriptive information about the 'Next and Previous' button. After activating it, screen reader should convey the information as 'Slide 2 Next button'.
Also, it should announce the image name or slide infromation when user activates the next and previous button.

**User Impact:**
Users who are dependent on screen readers are getting misguided if Screen reader (Narrator) is narrating incorrect information on invoking the 'Previous (<)' and 'Next (>)' buttons present in the "Hi there! I'm Icebreaker" card.

**WCAG Reference:**

[Understanding Success Criterion 1.3.1: Info and Relationships | WAI | W3C](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)

"Have feedback to share on Bugs? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-apoorva%40microsoft.com%7c551850210d444be9b28308d9691e3dde%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656400998658942%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=l6w0SvVcMSvyGnJ6%2B6uSSgd3braF2dZ7/AdBrqqzIR8%3D&reserved=0) “,

### Screenshots

https://github.com/microsoft/AdaptiveCards/assets/95913748/b4900f7a-5c1b-43a9-a4b8-15e3a4829d71

![MAS1 3 1-Screen Reader is not announcing the descriptive information while navigating to the 'Next or Previous' button](https://github.com/microsoft/AdaptiveCards/assets/95913748/4c325e3e-2a51-4531-aa33-16d5fd381b25)

### Card JSON

```JSON
{
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.2",
    "msTeams": {
        "width": "full"
    },
    "body": [
        {
            "type": "TextBlock",
            "text": "Hi there! I'm Icebreaker.",
            "wrap": true,
            "weight": "Bolder",
            "size": "Large"
        },
        {
            "type": "TextBlock",
            "text": "Hi IceBreaker A11y Team! Pranav . just added me to this team.",
            "wrap": true
        },
        {
            "type": "TextBlock",
            "text": "I'm Icebreaker, and I'll be helping you meet more people across your organization by randomly pairing you with someone new every week or month. You'll build rich connections, learn more about the people on this channel, and increase your social capital. Its a win-win-win situation!",
            "wrap": true
        },
        {
            "type": "TextBlock",
            "text": "Want to learn more about this application? Take a quick tour.",
            "wrap": true
        },
        {
            "type": "TextBlock",
            "text": "Great! Here are some of the things that Ice Breaker can help you with. Feel free to give it a try!",
            "wrap": true
        },
        {
            "type": "ColumnSet",
            "columns": [
                {
                    "type": "Column",
                    "verticalContentAlignment": "Center",
                    "width": "auto",
                    "items": [
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.ToggleVisibility",
                                    "iconUrl": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAS1BMVEX///8zMzMfHx+GhoYcHByIiIiEhIR+fn4XFxcgICDLy8sqKiotLS37+/soKCgkJCRdXV1sbGy3t7cNDQ3z8/PS0tKnp6dUVFSOjo78B79mAAACA0lEQVR4nO3da27CMBBFYUIepEALfXf/K22rCkj4W1mWzz3fCjzy2DOOQd5sJEmSJEmSJEmSJEmSJEkF7Hfn55fagyjpaz4cD8P5sfY4ihmH7tdpV3sgpYzb7s/rW+2hlNFfAuyGfe2xFDFdA+zm99qDKWG8BdgNtQdTQr8IcP6oPZoClgEOfe3RFNAPtwC3Y+3RFDDSZ3BaBjjVHk0BqxQlzuAqRfFrEJmi9AAny0Tj8Cm6DHCLDxC/BokzyG/VlilK3EVt1VqHr4O2aq3Dp2jUiZ6/BpEpmhQgP0WJAUalKDNAW7XG8U/0Sa0askzYqrVuimrViJsMPkW9fGkdfg3yWzUvXxpnq9Y6L19ah09R/Ik+q1VDpmhSgMg1aKvWul1SgMgysZ/ha3AzPV0DPBF30Z8kPVwCPH7WHksZ/DlcrUNmiPi9NODzYUBPE/D5Yn22eOCHmJCozBDx32nuvrXxZxGaqAFFA39vkZeoASHyExX5cdjDFENCA0f/XVteXeTvqJ76WxVQNBIaOPovbPIaOOYs4v9/GFE0AkJMWIs2cABhDRy1aHhtAxBQNLy2IQgrGtRTv9c2AAFr0WsbgrS6CN1RbeAAAopGQgPntQ1AQNHw2oYA/3ZewPuHqx11QL5hGfAOacBbsgHvAQe86RzwLrckSZIkSZIkSZIkSZIk/d83X50PsD0H9F4AAAAASUVORK5CYII=",
                                    "tooltip": "Previous",
                                    "targetElements": [
                                        "imageToToggle1",
                                        "imageToToggle2",
                                        "textToToogle1",
                                        "textToToogle1-1",
                                        "textToToggleWelcomeFooter",
                                        "textToToogle2-1",
                                        "textToToggleMatchFooter",
                                        "textToToogle2"
                                    ]
                                }
                            ]
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "auto",
                    "items": [
                        {
                            "type": "Image",
                            "url": "https://icebreaker123.blob.core.windows.net/images/teams2.png",
                            "id": "imageToToggle1",
                            "altText": "Expand your social circle"
                        },
                        {
                            "type": "Image",
                            "url": "https://icebreaker123.blob.core.windows.net/images/teams1.png",
                            "id": "imageToToggle2",
                            "isVisible": false,
                            "altText": "Setup a meeting with your new team mate"
                        }
                    ]
                },
                {
                    "type": "Column",
                    "width": "auto",
                    "verticalContentAlignment": "Center",
                    "items": [
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.ToggleVisibility",
                                    "iconUrl": "https://icons.veryicon.com/png/o/miscellaneous/commonly-used-icon-1/angle-brackets.png",
                                    "tooltip": "Next",
                                    "targetElements": [
                                        "imageToToggle1",
                                        "imageToToggle2",
                                        "textToToogle1",
                                        "textToToogle1-1",
                                        "textToToggleWelcomeFooter",
                                        "textToToogle2-1",
                                        "textToToggleMatchFooter",
                                        "textToToogle2"
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "type": "TextBlock",
            "text": "Expand your social circle",
            "wrap": true,
            "size": "Large",
            "weight": "Bolder",
            "id": "textToToogle1"
        },
        {
            "type": "TextBlock",
            "text": "Every week or month, I'll pair you with someone new. If you have the inclination that week, you can meet up, grab a coffee or lunch, and connect!",
            "wrap": true,
            "id": "textToToogle1-1"
        },
        {
            "type": "TextBlock",
            "text": "Look out for another message in which I'll pair you with your first match.",
            "wrap": true,
            "id": "textToToggleWelcomeFooter"
        },
        {
            "type": "TextBlock",
            "text": "Setup a meeting with your new team mate!",
            "wrap": true,
            "size": "Large",
            "weight": "Bolder",
            "id": "textToToogle2",
            "isVisible": false
        },
        {
            "type": "TextBlock",
            "text": "After matched with a teammate, you will be given option to set up a meeting or chat with your co-worker!",
            "wrap": true,
            "id": "textToToogle2-1",
            "isVisible": false
        },
        {
            "type": "TextBlock",
            "text": "Pause matches whenever you need. Too busy that week? Out of the office? No problem. Just click the button at anytime. Don't forget to un-pause when you're back",
            "wrap": true,
            "id": "textToToggleMatchFooter",
            "isVisible": false
        },
        {
            "type": "ColumnSet",
            "columns": [
                {
                    "type": "Column",
                    "items": [
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.OpenUrl",
                                    "title": "Open App",
                                    "url": "${openAppURL}"
                                }
                            ]
                        }
                    ],
                    "width": "120px"
                },
                {
                    "type": "Column",
                    "items": [
                        {
                            "type": "ActionSet",
                            "actions": [
                                {
                                    "type": "Action.OpenUrl",
                                    "title": "Configure",
                                    "url": "${configureUrl}"
                                }
                            ]
                        }
                    ],
                    "width": "150px"
                }
            ]
        }
    ]
}
```

**Comment by anna-dingler** [comment]
Hi @vagpt,

This narrator behavior appears to be a card authoring issue. In schema 1.6, we introduced a Carousel element that properly handles this behavior. Action.ToggleVisibility should not be used for this scenario.

Let me know if you have any questions!
